### DISCUSSION TITLE
Units of the drag parameter

### DISCUSSION THREAD

**Discussion by ThomasJannaudCAM** [idefix-code/idefix#282] [General]
Hi all,

Quick question that has been bugging me for a bit.
In the documentation of the dust module (https://idefix.readthedocs.io/latest/modules/dust.html) the physical meaning of the drag parameters is explained, relating it to the sound speed, the gas mean free path and sound speed, the grain stopping time, size and density. But there are no units mentioned.
I have assumed that it is all in CGS units, and it seems confirmed by the comments in the FargoPlanet setup, but I wanted to be sure that I am not missing anything.

Thanks in advance,

Best wishes,

Thomas

**Comment by glesur** [comment]
There is no unit, because it's in code units :-) So no, it doesn't assume CGS. The only thing idefix uses is `gamma_i`, and this is units of 1/density units*time units, so whatever density units and time units you choose, gamma will be in those units. Now how you construct gamma from your favourite grain model is up to you, but in the end, it should be put in code units (i.e. your choice of density unit, time unit, length unit)

**Reply by ThomasJannaudCAM** [reply]
Hi Geoffroy,

Thanks, I realized my mistake shortly after, I should have reread my slides from the idefix-days.
One of the correct ways to do this is to go through the adimensioned Stokes number St.
For the size drag law for instance St = beta Omega/rho/c_s
If you then assume the Epstein regime St = rho_s a Omega/rho/c_s
The input parameters set beta=c_s/gamma, Omega, rho and c_s in code units, so set the adimensioned Stokes number St.
You then set a normalization (value in physical units) for rho_s, Omega, rho and c_s. This then gives you a value in physical units for a.
(Notations are the same as in the documentation)

**Reply by birnstiel** [reply]
If I may add here: the definition of the Stokes number in the Epstein regime is incorrect. This is also a tiny mistake in the Johansen paper you refer to [here](https://github.com/idefix-code/idefix/blob/d0e82202ca90d0b664c71582962c2ae255bc6528/test/Dust/FargoPlanet/setup.cpp#L43). The velocity in the drag formula should be the mean thermal velocity $v_{th} = \sqrt{8/\pi} \cdot c_s$, not just $c_s$.

**Reply by ThomasJannaudCAM** [reply]
Oh I did not know this, thank you !

**Reply by neutrinoceros** [reply]
Off topic, but I'm sure you guys will love to learn that GitHub markdown actually supports a subset of LaTeX : https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/writing-mathematical-expressions

**Reply by birnstiel** [reply]
I did use it in my answer already 😃

**Reply by ThomasJannaudCAM** [reply]
Yes and I was wondering how you did that, thanks for the tip Clément :)

**Reply by neutrinoceros** [reply]
Indeed you did... 😅
I just learned that it doesn't render properly on GitHub Mobile. 🤦🏻‍♂️
